### PR TITLE
[FX-1396] Get rid of `AbstractForageElement` in favor of interfaces

### PR DESCRIPTION
**Pull request by devinmorgan**
> 📢 See individual commits for a detailed, step-by-step walkthrough of the changes

## What (and Why)
* Remove `ForageSDKInterface`: This interface now unnecessarily constrains/couples the `ForageSDK` and `ForageTerminaSDK`, which is problematic given the diverging requirements between Ecom (EditText-only UI, redundant Vaults) and POS (EditText and Keypad UI, no vault redundancy) flows.
    - As a result, all `ForageSDK` "params" get moved to `ForageSDK.kt`
* Make VaultWrapper hierarchy more generic: adds an `EnvConfig` param to `getVaultSubmitter` to support POS-specific flows wherein UI elements do not have an EnvConfig bound to them.
* Breakup `AbstractForageElement` into interfaces: Provide necessary flexibility for non-EditText-based UI components 
     - by splitting up `AbstractForageElement` class into these atomic interfaces: 
        - `DynamicEnvElement`
        - `ForageElement`
        - `EditTextElement`
    - moving `ForageConfig` initialization logic to a separate Manager class

## Why
See above

## Test Plan
- ✅ Added tests for standalone `ForageConfigManager` and updates existing test suite
- ❌ No need to manually test; passing CI tests is sufficient!

## Demo
N/A since this is an implementation detail

## How
Can be rolled out once #253 is merged!

### CODE REVIEW

**Comment by devinmorgan** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/teamforage/forage-android-sdk/254?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#256** <a href="https://app.graphite.dev/github/pr/teamforage/forage-android-sdk/256?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#255** <a href="https://app.graphite.dev/github/pr/teamforage/forage-android-sdk/255?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#254** <a href="https://app.graphite.dev/github/pr/teamforage/forage-android-sdk/254?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#253** <a href="https://app.graphite.dev/github/pr/teamforage/forage-android-sdk/253?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#252** <a href="https://app.graphite.dev/github/pr/teamforage/forage-android-sdk/252?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @devinmorgan and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: forage-android/src/main/java/com/joinforage/forage/android/core/ui/element/ForageElement.kt [hunk 1]
```diff
@@ -24,22 +25,11 @@ import com.joinforage.forage.android.core.ui.element.state.ElementState
...
- *
- * @property typeface The [Typeface](https://developer.android.com/reference/android/graphics/Typeface)
- * that is used to render text within the ForageElement.
- * @see * [Online-only Android Quickstart](https://docs.joinforage.app/docs/forage-android-quickstart)
- * * [POS Terminal Android Quickstart](https://docs.joinforage.app/docs/forage-terminal-android)
- * * [Guide to styling Forage Android Elements](https://docs.joinforage.app/docs/forage-android-styling-guide)
- */
-interface ForageElement<T : ElementState> {
-    var typeface: Typeface?
+interface DynamicEnvElement {
```
Comment (djoksimo) [inline forage-android/src/main/java/com/joinforage/forage/android/core/ui/element/ForageElement.kt:32]: Can this be internal?

**Review by djoksimo — CHANGES_REQUESTED** [review]

File: forage-android/src/main/java/com/joinforage/forage/android/core/ui/element/ForageElement.kt (hunk 1 above)
Comment (devinmorgan) [reply forage-android/src/main/java/com/joinforage/forage/android/core/ui/element/ForageElement.kt:32]: Oooooo and, on that note, we can actually make all of these internal:
* `DynamicElement`
* `EditTextElement`
* `ForageElement<T : ElementState>`
which would be a nice win for reducing our public contract. Great callout, Danilo!

Comment (devinmorgan) [reply forage-android/src/main/java/com/joinforage/forage/android/core/ui/element/ForageElement.kt]: OK - all three have been marked `internal` with the last commit!

**Comment by devinmorgan** [comment]
### Merge activity

* **Jun 17, 10:05 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/teamforage/forage-android-sdk/254) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Review by djoksimo — APPROVED** [review]

**Pull request merged**